src/main/java/edu/byu/cs/autograder/QualityAnalyzer.java

```diff
@@ -64,82 +66,68 @@ private QualityAnalysis parseAnalysis(String output) {
+                if (errors.containsKey(reporter)) {
+                    errors.get(reporter).forEach(s -> reporterResultsBuilder.append("\t\t").append(s).append("\n"));
+                }
+                if (!reporterResultsBuilder.isEmpty()) {
+                    categoryResultsBuilder.append("\t").append(reporter).append(":\n")
+                            .append(reporterResultsBuilder);
                 }
             }
             if (!categoryResultsBuilder.isEmpty()) {
                 resultsBuilder.append(category.name()).append(":\n").append(categoryResultsBuilder);
             }
         }
 
+        if(!warnings.isEmpty()) {
+            resultsBuilder.append("Warnings:\n");
+            warnings.forEach(s -> resultsBuilder.append("\t").append(s).append("\n"));
+        }
+
         if (resultsBuilder.isEmpty()) resultsBuilder.append("Good job!");
         else resultsBuilder.deleteCharAt(resultsBuilder.length() - 1);
         return resultsBuilder.toString();
     }
 
-    private String getNotes(QualityAnalysis analysis) {
+    private String getNotes() {
         StringBuilder builder = new StringBuilder();
         for (QualityRubricCategory category : qualityRubricItems.categories()) {
-            float categoryPoints = getCategoryPoints(analysis, category);
-            builder.append(category.name()).append(": ").append(categoryPoints).append("/").append(category.value())
-                    .append("\n");
+            boolean categoryPasses = categoryPasses(category);
+            builder.append((categoryPasses) ? "✓" : "✗").append(" ").append(category.name()).append("\n");
         }
         return builder.toString();
     }
 
 
-    private float getCategoryPoints(QualityAnalysis analysis, QualityRubricCategory category) {
-        float deductions = 0;
-        float totalWeight = 0;
-        for (QualityRubricItem item : category.items()) {
-            for (QualityRubricReporter reporter : item.reporters()) {
-                float weight = item.weight() * reporter.weight();
-                totalWeight += weight;
-                if (analysis.errors().containsKey(reporter.name())) {
-                    deductions += weight;
-                }
-            }
+    private boolean categoryPasses(QualityRubricCategory category) {
+        for(String reporter : category.reporters()) {
+            if(errors.containsKey(reporter)) return false;
         }
-        return category.value() * (1 - (deductions / totalWeight));
+        return true;
     }
 
 
-    public record QualityOutput(float score, String results, String notes) {}
-
-    private record QualityAnalysis(Map<String, List<String>> errors, List<String> warnings) {}
-
-    private record QualityRubricReporter(String name, int weight) {}
-
-    private record QualityRubricItem(Set<QualityRubricReporter> reporters, int weight) {}
+    public record QualityAnalysis(float score, String results, String notes) {}
```

Comment:
Would this be an appropriate place to use JavaDoc comments to describe these parameters? The `score` value is pretty clear, but I'm actually not sure about the difference between `results` and `notes`. Some defining documentation could help prevent confusion in the future.

Reply:
The new documentation is most helpful!